frb_rust/src/macros.rs

```diff
@@ -72,30 +72,35 @@ macro_rules! transfer {
 /// onmessage.forget();
 /// let port2 = channel.port2();
 /// // Declare the transferable with the same name and type
-/// flutter_rust_bridge::spawn!(|port2: MessagePort| {
+/// flutter_rust_bridge::spawn!(4, |port2: MessagePort| {
 ///     port2.post_message(&JsValue::from(true)).unwrap();
 /// });
 /// # } #[cfg(not(target_family = "wasm"))] fn main() {}
 /// ```
 #[macro_export]
 macro_rules! spawn {
-    ($($tt:tt)*) => {{
+    ($wc:expr, $($tt:tt)*) => {{
         let worker = $crate::transfer!($($tt)*);
         #[cfg(not(target_family = "wasm"))]
         {
-            $crate::thread::THREAD_POOL.lock().execute(worker)
+            use $crate::thread;
+            thread::create_thread_pool($wc);
```

Comment:
The problem of the implementation of the PR is that, it seems the `worker_count` only works for the *first* call of `spawn!`, since `create_thread_pool` only creates the pool in the first call. This will make users very confused IMHO :(

Reply:
Oh, I thought that calling spawn! was reposible for creating the pool itself. I now see that it is for passing functions into the existing thread pool. Thanks for telling me:)